geode-core/src/test/java/org/apache/geode/management/internal/cli/commands/ConfigurePDXCommandTest.java

```diff
@@ -0,0 +1,379 @@
+    // Custom Configuration as False
+    gfshParserRule
+        .executeAndAssertThat(command, BASE_COMMAND_STRING + "--ignore-unread-fields=false")
+        .statusIsSuccess().hasNoFailToPersistError()
+        .hasReadSerializedConfigured(DEFAULT_PDX_READ_SERIALIZED, cacheCreation)
+        .hasIgnoreUnreadFieldsConfigured(false, cacheCreation)
+        .hasPersistenseConfigured(DEFAULT_PDX_PERSISTENT, null, cacheCreation);
+
+    verify(cacheCreation, times(0)).setPdxSerializer(any());
+    verify(command, times(1)).persistClusterConfiguration(any(), any());
+    verify(command, times(0)).createReflectionBasedAutoSerializer(anyBoolean(), any());
+  }
+
+  @Test
+  public void executionShouldCorrectlyConfigurePortableAutoSerializableClassesWhenUsingCustomPattern() {
+    String[] patterns = new String[] {"com.company.DomainObject.*#identity=id"};
+
+    // Custom Settings
+    gfshParserRule
+        .executeAndAssertThat(command,
+            BASE_COMMAND_STRING + "--portable-auto-serializable-classes=" + patterns[0])
+        .statusIsSuccess().hasNoFailToPersistError()
+        .hasReadSerializedConfigured(DEFAULT_PDX_READ_SERIALIZED, cacheCreation)
+        .hasIgnoreUnreadFieldsConfigured(DEFAULT_PDX_IGNORE_UNREAD_FIELDS, cacheCreation)
+        .hasPersistenseConfigured(DEFAULT_PDX_PERSISTENT, null, cacheCreation)
+        .containsOutput("Portable Classes = [com.company.DomainObject.*#identity=id]")
+        .containsOutput("PDX Serializer = org.apache.geode.pdx.ReflectionBasedAutoSerializer");
+
+    verify(cacheCreation, times(1)).setPdxSerializer(any());
+    verify(command, times(1)).persistClusterConfiguration(any(), any());
+    verify(command, times(1)).createReflectionBasedAutoSerializer(true, patterns);
+  }
+
+  @Test
+  public void executionShouldCorrectlyConfigureAutoSerializableClassesWhenUsingCustomPattern() {
+    String[] patterns = new String[] {"com.company.DomainObject.*#identity=id"};
+
+    // Custom Settings
+    gfshParserRule
+        .executeAndAssertThat(command,
+            BASE_COMMAND_STRING + "--auto-serializable-classes=" + patterns[0])
+        .statusIsSuccess().hasNoFailToPersistError()
+        .hasReadSerializedConfigured(CacheConfig.DEFAULT_PDX_READ_SERIALIZED, cacheCreation)
+        .hasIgnoreUnreadFieldsConfigured(CacheConfig.DEFAULT_PDX_IGNORE_UNREAD_FIELDS,
+            cacheCreation)
+        .hasPersistenseConfigured(CacheConfig.DEFAULT_PDX_PERSISTENT, null, cacheCreation)
+        .containsOutput("Non Portable Classes = [com.company.DomainObject.*#identity=id]")
+        .containsOutput("PDX Serializer = org.apache.geode.pdx.ReflectionBasedAutoSerializer");
+
+    verify(cacheCreation, times(1)).setPdxSerializer(any());
+    verify(command, times(1)).persistClusterConfiguration(any(), any());
+    verify(command, times(1)).createReflectionBasedAutoSerializer(false, patterns);
+  }
+
+  static class CustomGfshParserRule extends GfshParserRule {
```

Comment:
instead of adding this class, you can just add a method in CommandResultAssert to expose the commandResult there. We didn't do that because we hadn't had a need to, but now that you need it to do custom assertion. We should just add the method getCommandResult.

Reply:
@jinmeiliao: the `CommandResultAssert` class already exposed the method `getCommandResult()` when I wrote the test, I needed something different.
I basically wanted a `PDXCommandResultAssert` (just changed the name in my last commit so it makes more sense), internal to the test class, so I can assert different functional things related to the PDX configuration command in just one invocation (`hasPersistenceConfigured`, `hasDefaultsConfigured`, etc.) without having to repeat the assertions in every single test method; it looks more readable and natural.
The method `GfshParserRule.executeAndAssertThat()` returns an instance of `CommandResultAssert` but I want it to return an instance of my `PDXCommandResultAssert` assertion, that's why I extended the class in the first place.
Do you see any complications with this approach?.
Cheers.